.azure-pipelines/main-pipeline.yml

```diff
@@ -116,15 +116,27 @@ stages:
               vsixFile: '$(Pipeline.Workspace)/$(artifactPath)/$(packagedTask)'
               extensionTag: '$(devExtensionTag)'
               extensionName: '$(extensionName) DEV'
-              extensionVersion: '$(queryDevVersion.Extension.Version)'
-              updateTasksId: true
               extensionVisibility: 'private'
               extensionPricing: 'free'
               shareWith: 'datadog-ci'
+              extensionVersion: '$(queryDevVersion.Extension.Version)'
+              updateTasksId: true # The DEV task ID is overridden to 99cf84cd-0117-58b7-b1dd-d95e5ae9baf8 (based on the `extensionTag`)
```

Comment:
The ID of the task that belongs to the DEV extension is **constant**. I found its value using:

```bash
npx tfx-cli build tasks list --service-url https://dev.azure.com/datadog-ci --json | jq '.[] | select(.name == "SyntheticsRunTests") | [.id, .name, .version, .contributionVersion]'
```

The result of this command - today - is:

```json
[
  "99cf84cd-0117-58b7-b1dd-d95e5ae9baf8", // DEV task id
  "SyntheticsRunTests",
  { "major": 0, "minor": 1, "patch": 102, "isTest": false }, // DEV task version
  "0.0.102" // DEV extension version
]
[
  "60b18503-c6d6-4e4b-a6b2-52fc6fb3d525", // Public task ID
  "SyntheticsRunTests",
  { "major": 0, "minor": 1, "patch": 4, "isTest": false }, // Public task version
  "1.4.0" // Public extension version
]
```

Reply:
How do you know it's constant?

Reply:
It's documented [here](https://github.com/microsoft/azure-devops-extension-tasks/blob/982a9ba89cdded9c814a67baf9541d67606b0b36/BuildTasks/PublishExtension/v4/task.json#LL200C51-L200C51), and I also looked at the code: here they [gather the components to generate the ID](https://github.com/microsoft/azure-devops-extension-tasks/blob/982a9ba89cdded9c814a67baf9541d67606b0b36/BuildTasks/Common/v4/Common.ts#L434-L439), and here they [use `uuidv5` to generate the constant task ID](https://github.com/microsoft/azure-devops-extension-tasks/blob/982a9ba89cdded9c814a67baf9541d67606b0b36/BuildTasks/Common/v4/Common.ts#L337-L338).

The only part that makes the ID change is the `extensionTag` that we add for the DEV extension only (its value is `-dev`):

https://github.com/DataDog/datadog-ci-azure-devops/blob/65ef41e1b92dd919a922dae169d216b44063a5fd/azure-pipelines.yml#L117

In consequence, the task has to be referenced with this full task "public" identifier ([example in our E2E tests](https://github.com/DataDog/datadog-ci-azure-devops/blob/65ef41e1b92dd919a922dae169d216b44063a5fd/ci/e2e.azure-pipelines.yml#L30C27-L30C41)):

```
Datadog.datadog-ci-dev.synthetics-application-testing.SyntheticsRunTests@0
```